dnsapi/dns_digitalocean.sh

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env sh
+
+## Will be called by acme.sh to add the txt record to your api system.
+## returns 0 means success, otherwise error.
+
+## Author: thewer <github at thewer.com>
+## GitHub: https://github.com/gitwer/acme.sh
+
+##
+## Environment Variables Required:
+##
+## DO_API_KEY="75310dc4ca779ac39a19f6355db573b49ce92ae126553ebd61ac3a3ae34834cc"
+##
+## DO_DOMAIN_START="3"
+## start of the digital ocean dns base domain from the LEFT
+## (EG: one.two.three.four.five.com -> one.two & three.four.five.com)
+##
+
+#####################  Public functions  #####################
+
+## Create the text record for validation.
+## Usage: fulldomain txtvalue
+## EG: "_acme-challenge.www.other.domain.com" "XKrxpRBosdq0HG9i01zxXp5CPBs"
+dns_digitalocean_add() {
+  fulldomain=$1
+  txtvalue=$2
+  _info "Using digitalocean dns validation - add record"
+  _debug fulldomain "$fulldomain"
+  _debug txtvalue "$txtvalue"
+  _debug DO_DOMAIN_START "$DO_DOMAIN_START"
+
+  ## split the domain for DO API
+  if ! _get_base_domain "$fulldomain" "$DO_DOMAIN_START"; then
+    _err "invalid domain or split"
+    return 1
+  fi
+  _debug _sub_domain "$_sub_domain"
+  _debug _domain "$_domain"
+
+  ## Set the header with our post type and key auth key
+  export _H1="Content-Type: application/json"
+  export _H2="Authorization: Bearer $DO_API_KEY"
+  PURL='https://api.digitalocean.com/v2/domains/'$_domain'/records'
```

Comment:
Please call `_saveaccountconf DO_API_KEY "$DO_API_KEY"` to save the key for future automatic renewal.